docs/HelloWorldWalkthrough.md

```diff
@@ -0,0 +1,312 @@
+    ```
+
+## Solution structure
+This sample will create a Host app, a DLL project and include a solution level file as follows:
+- MyHostApp.sln
+  - MySecretEnclave.edl
+  - MyHostApp.vcxproj
+    - main.cpp
+  - MySecretEnclave.vcxproj
+    - MySecretEnclaveExports.cpp
+
+## Build the sample
+### Create the initial host app project and solution
+  1. Start VS, create new C++ Console App named **MyHostApp.vcxproj**
+### Add the Enclave Definition Language file to the solution
+  2. Add the Enclave Definition Language (EDL) file to the solution where you define your interface.
+     * Right click on solution, add new item, text file, named **MySecretEnclave.edl**
+     * Define a simple method for the enclave:
+        ```c
+        enclave
+        {
+            trusted
+            {
+                uint32_t DoSecretMath(
+                uint32_t val1,
+                uint32_t val2
+                );
+            };
+        };
+        ```
+### Add the Enclave DLL project to the solution
+Add a new DLL project and to the solution and configure it. This is the enclave dll.
+1. Right click add project, DLL Project, name it **MySecretVBSEnclave.vcxproj**
+2. Add references to the two nuget packages you created when you built the repro, located under \<repo root\>\\_build folder. Right-Click on the project, choose Manage Nuget packages – add a new package source for the _build folder where these were placed, and install the packages:
+    * Microsoft.Windows.VbsEnclave.SDK.0.0.0.nupkg
+    * Microsoft.Windows.VbsEnclave.CodeGenerator.0.0.0.nupkg
+3. Update DLLMain.cpp with the enclave configuration information as follows. For more information on these values and how they impact enclave sealing policy, please refer to https://learn.microsoft.com/en-us/windows/win32/api/ntenclv/ne-ntenclv-enclave_sealing_identity_policy
+    ```c
+    #include "pch.h"
+    #include <array>
+
+    // Family ID will be assigned via Azure Trusted Signing
+    // For private testing you can use any 16 byte value except 1 or 0
+    #define SAMPLE_ENCLAVE_FAMILY_ID \
+        { \
+            0xED, 0X1D, 0xD0, 0x21, 0xC1, 0xB3, 0x42, 0x4C, \
+            0x96, 0x49, 0xF6, 0xE9, 0x18, 0x18, 0x70, 0x36, \
+        }
+    
+    #define SAMPLE_ENCLAVE_IMAGE_ID \
+        { \
+            0x9B, 0x9B, 0x50, 0xDD, 0x83, 0x2F, 0x44, 0xFD, \
+            0xB3, 0x8D, 0xAD, 0x87, 0x92, 0xD6, 0x9F, 0x42, \
+        }
+    
+    // Example version - 10.0.26100.0 -> A.0.65F4.00
+    #define ENCLAVE_IMAGE_VERSION 0xA065F400 
+
+    //Security version number
+    #define SAMPLE_ENCLAVE_SVN 1000
+    
+    #define ENCLAVE_ADDRESS_SPACE_SIZE \
+        0x20000000          // The expected virtual size of the private address range 
```

Comment:
>// The  [](http://example.com/codeflow?start=28&length=7)

VBS enclaves must be a multiple of 2 MB in size.

[https://learn.microsoft.com/en-us/windows/win32/api/enclaveapi/nf-enclaveapi-createenclave](https://learn.microsoft.com/en-us/windows/win32/api/enclaveapi/nf-enclaveapi-createenclave)